File: 88. Merge Sorted Array/main.cpp
// Solution By Vandit Jain (jainvandit99)
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int i = nums1.size()-n-1;
        int j = nums2.size() - 1;
        int k = nums1.size() - 1;
        while(i>=0 && j >=0){
            if(nums1[i] > nums2[j]){
                nums1[k--] = nums1[i--];
            }else{
                nums1[k--] = nums2[j--];
            }
        }
        while(j>=0){
            nums1[k--] = nums2[j--];
        }
    }
};

File: 88. Merge Sorted Array/main.py
"""
Author: Amal Matty Antony
"""
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        ptr = m+n-1
        m-=1
        n-=1
        while(n>=0 and m >=0):
            if nums1[m] > nums2[n]:
                nums1[ptr] = nums1[m]
                m-=1
            else:
                nums1[ptr] = nums2[n]
                n-=1
            ptr-=1
        while(n>=0):
            nums1[ptr] = nums2[n]
            ptr-=1
            n-=1
        